backend/src/apiserver/client/subject_access_review_fake.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Arrikto Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	authzv1 "k8s.io/api/authorization/v1"
+)
+
+type FakeSubjectAccessReviewClient struct {
+}
+
+func (FakeSubjectAccessReviewClient) Create(*authzv1.SubjectAccessReview) (*authzv1.SubjectAccessReview, error) {
+	return &authzv1.SubjectAccessReview{Status: authzv1.SubjectAccessReviewStatus{
+		Allowed:         true,
+		Denied:          false,
+		Reason:          "",
+		EvaluationError: "",
+	}}, nil
+}
+
+func (FakeSubjectAccessReviewClient) CreateContext(context.Context, *authzv1.SubjectAccessReview) (*authzv1.SubjectAccessReview, error) {
```

Comment:
Have you considered making a thin interface for this client, so that we do not need to expose this method we don't use.
example: https://github.com/kubeflow/pipelines/blob/1165ed243b381ae37a72e245d685ba318d58b345/backend/src/apiserver/client/argo.go#L28-L34

Reply:
Actually where does this `CreateContext` come from? I don't see it being asked by the interface:
https://godoc.org/k8s.io/client-go/kubernetes/typed/authorization/v1#SubjectAccessReviewInterface

Reply:
We are using client-go v0.17.9: https://pkg.go.dev/k8s.io/client-go@v0.17.9/kubernetes/typed/authorization/v1#SelfSubjectAccessReviewInterface

I wrote a custom `SubjectAccessReviewInterface` which we can remove it in the future, when we upgrade `client-go`.